### PR TITLE
Offering some cleanup and starting to add features

### PR DESCRIPTION
**Pull request by icefoxen**
Heyo,

You hardly have to accept this particular pull request but I want to make you aware of it and see how you like it.  It mostly contains a bunch of error-handling and a little bit of refactoring, making everything use Result<> instead of .unwrap().  The exception is the last commit, which has the start of a unified filesystem module.

I intend to keep working on this fork as long as my interest holds (which usually means about two weeks), and use Love2D as a rough API model (as you can see in filesystem.rs).  No idea what I'll end up fiddling with next, but at a guess: audio, fonts, image/drawing, and input.

Cheers!

### CODE REVIEW

**Comment by svenstaro** [comment]
Thanks for these! I will take a proper look soon.

**Comment by icefoxen** [comment]
Well I fail at git for putting everything on the master branch instead of making a branch for this pull request.  So it contains ALL my in progress work.  Oh well.

**Comment by svenstaro** [comment]
Holy crap dude, you're on fire.

**Comment by icefoxen** [comment]
Don't worry, it won't last.  By the end of this week I expect to have burned out on this project and moved on to something new and shiny.

File: Cargo.toml [hunk 1]
```diff
@@ -13,11 +18,15 @@ name = "ggez"
 path = "src/lib.rs"
 
 [dependencies]
-sdl2 = "0.18"
-sdl2_image = "0.18"
-sdl2_mixer = "0.17"
-sdl2_ttf = "0.16"
-rand = "0.3"
+sdl2 = "^0.23"
```
Comment (icefoxen) [inline Cargo.toml]: I honestly did not realize that!  I'll fix it.

**Comment by icefoxen** [comment]
Just gonna close this and break it into smaller pieces.

**Pull request closed without merging**